### PR TITLE
Session5

### PR DESCRIPTION
**Pull request by ankita1696**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
-const heading2 = React.createElement("h2",{},"Hello React");
+import React from "react";
+import ReactDOM from "react-dom/client";
```
Comment (nitishnivedan) [inline src/App.js:2]: can give an empty line between direct imports and file imports for better readability

File: src/Components/Header.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import teamlogo from "../asset/teamlogo.jpg";
+import user from "../asset/user.png";
+
+const HeaderComponent = () => {
+  return (
+    <div className="header">
+      <Logo></Logo>
+      <SearchBar></SearchBar>
```
Comment (nitishnivedan) [inline src/Components/Header.js]: SearchBar is neither imported nor present in this file. Is this not throwing error. or Am i missing anything here.

File: src/Components/Team/Card.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+const CardComponent = ({ team }) => {
```
Comment (nitishnivedan) [inline src/Components/Team/Card.js]: can destructure further. const CardComponent = ({ team: {name,location}}) => (

>

File: src/Components/Team/Team.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  ));
+};
+
+const TeamBody = () => {
+  const [filteredResult, setFilteredResult] = useState(teamData);
+  return (
+    <>
+      <SearchBar setFilteredResult={setFilteredResult}></SearchBar>
+      <div className="card-container">
+        <CardContainer filteredResult={filteredResult} />
```
Comment (nitishnivedan) [inline src/Components/Team/Team.js:36]: can use a key prop here and assign a unique value for better performance.

File: src/Components/SearchBar.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
```
Comment (smriti-porwal) [inline src/Components/SearchBar.js:1]: Can remove this. It is not getting used here.

File: src/App.css [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  background: transparent;
+  margin: 0;
+  padding: 10px;
+  font-size: 14px;
+  color: inherit;
+  border: 1px solid grey;
+  border-radius: 5px;
+}
+
+input[type="search"]::placeholder {
```
Comment (ap221882) [inline src/App.css:37]: Use html tag- css selectors on the top of your code, it will prevent CSS specificity collisions

File: src/App.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
...
+import TeamComponent from "./Components/Team/Team";
+
 const root2 = ReactDOM.createRoot(document.getElementById("root2"));
-root2.render(heading2)
\ No newline at end of file
+root2.render(<TeamComponent></TeamComponent>);
```
Comment (ap221882) [inline src/App.js:7]: No need for a Closing tag. You can write <TeamComponent/>

File: src/Components/Header.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      <User></User>
+    </div>
+  );
+};
+
+const Logo = () => {
+  return (
+    <div className="wrapper_logo">
+      <img className="header_logo" src={teamlogo}></img>
```
Comment (ap221882) [inline src/Components/Header.js:17]: img is a self closing html tag, you can write

File: src/Components/SearchBar.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import { useState } from "react";
+import teamData from "./Team/teamData";
+
+const searchMember = (searchText) => {
+  // Logic for searching inside data
+  return teamData.filter(
+    (res) =>
+      res.name.toLocaleLowerCase().includes(searchText.toLocaleLowerCase()) ||
```
Comment (ap221882) [inline src/Components/SearchBar.js:9]: You can use a formatter for uniform formatting. Increases readability.

**Review by ap221882 — COMMENTED** [review]
Nicely written. You can make separate folders for components and put their logic and styles there only.

File: src/Components/Header.js (hunk 2 above)
Comment (ankita1696) [reply src/Components/Header.js]: Actually it was for 4th assignment moved the header inside body now. Forgot to remove it

File: src/Components/Team/Team.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+const TeamHeader = () => {
+  return (
+    <div className="wrapper_logo">
+      <img className="header_logo" src={logo}></img>
+    </div>
+  );
+};
+
+const CardContainer = ({ filteredResult }) => {
```
Comment (roshantrivedi) [inline src/Components/Team/Team.js:24]: can make separate file for different containers. It will make Code refactoring easier in future.

**Pull request merged**